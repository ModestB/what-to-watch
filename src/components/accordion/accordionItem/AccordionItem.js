import React from "react";

import CloseIcon from "../../../icons/js/Close";

import classes from "./AccordionItem.module.scss";

export const AccordionItem = (props) => {
  return (
    <section className={`${classes.item}`}>
      <header
        className={`${classes.header} ${props.show ? classes.active : ""}`}
        onClick={() => props.clickHandler(props.index, props.title)}
      >
        <h2>
          <div className={`${classes.title}`}>
            <p>{props.title}</p>
          </div>
        </h2>
        <div className={`${classes.headerOverlay}`}></div>
      </header>

      <div
        className={`${classes.body}  ${
          props.index === 0 ? classes.bodyLeft : classes.bodyRight
        } ${props.show ? classes.active : ""}`}
      >
        <CloseIcon
          className={`${classes.close} icon-close icon-close--dark`}
          onClick={() => props.clickHandler(props.index, props.title)}
          width="20px"
          height="20px"
        />
        <div>{props.body}</div>
      </div>
    </section>
  );
};

export default AccordionItem;
