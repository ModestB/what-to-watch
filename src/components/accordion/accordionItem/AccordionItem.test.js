import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { AccordionItem } from "./AccordionItem";

configure({ adapter: new Adapter() });

describe("<AccordionItem />", () => {
  let wrapper;
  let props = {
    title: "Trailers",
    body: <p>No Information</p>,
  };

  beforeEach(() => {
    wrapper = shallow(<AccordionItem {...props} />);
  });

  it("should match snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match props title in header", () => {
    expect(wrapper.find(".title p").text()).toEqual(props.title);
  });
});
