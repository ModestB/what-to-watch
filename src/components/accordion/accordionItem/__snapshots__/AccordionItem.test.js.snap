// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> should match snapshot 1`] = `
"<section className=\\"item\\">
  <header className=\\"header \\" onClick={[Function: onClick]}>
    <h2>
      <div className=\\"title\\">
        <p>
          Trailers
        </p>
      </div>
    </h2>
    <div className=\\"headerOverlay\\" />
  </header>
  <div className=\\"body  bodyRight \\">
    <SvgClose className=\\"close icon-close icon-close--dark\\" onClick={[Function: onClick]} width=\\"20px\\" height=\\"20px\\" />
    <div>
      <p>
        No Information
      </p>
    </div>
  </div>
</section>"
`;
